### ISSUE TITLE
Old user data accessible : security issue

### ISSUE THREAD

**Issue by dayitv89** [uber/rides-ios-sdk#101]
User token stored in keychain and when app deleted and again install, ride sdk read user data from keychain and old user data is accessible. What if user token expires it request for re-auth? 

User token can be deleted by the `TokenManager.deleteToken()` only. 

Why not store a variable in user-default for checking first run, and if the variable not found clean auth token, insist of show old user data.

**Comment by jbrophy17** [comment]
Yes this is true, NSKeyedArchiver does not delete data when an app is uninstalled. Other apps cannot access this data, but I agree it would be useful to have it clear on a fresh install.

**Comment by dayitv89** [comment]
Have a look to PR #102 for this.
